command/agentproxyshared/cache/lease_cache.go

```diff
@@ -267,15 +336,25 @@ func (c *LeaseCache) Send(ctx context.Context, req *SendRequest) (*SendResponse,
 		inflight.remaining.Inc()
 		defer inflight.remaining.Dec()
 
-		c.inflightCache.Set(id, inflight, gocache.NoExpiration)
-		idLock.Unlock()
+		c.inflightCache.Set(staticSecretCacheId, inflight, gocache.NoExpiration)
+		idLockStaticSecret.Unlock()
 
 		// Signal that the processing request is done
 		defer close(inflight.ch)
 	}
 
-	// Check if the response for this request is already in the cache
-	cachedResp, err := c.checkCacheForRequest(id)
+	// Check if the response for this request is already in the dynamic secret cache
+	cachedResp, err := c.checkCacheForDynamicSecretRequest(dynamicSecretCacheId)
```

Comment:
There's a typo here, should be staticSecretCacheId. It'll be fixed in the next PR.